pkg/actors/actors.go

```diff
@@ -1153,19 +1153,38 @@ func (a *actorsRuntime) ExecuteLocalOrRemoteActorReminder(ctx context.Context, r
 
 	// If the reminder was cancelled, delete it.
 	if errors.Is(err, ErrReminderCanceled) {
+		errCh := make(chan error, 1)
 		go func() {
+			defer close(errCh)
+
 			log.Debugf("Deleting reminder which was cancelled: %s", reminder.Key())
-			reqCtx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+			reqCtx, cancel := context.WithTimeout(ctx, time.Second*15)
 			defer cancel()
+
 			if derr := a.DeleteReminder(reqCtx, &DeleteReminderRequest{
 				Name:      reminder.Name,
 				ActorType: reminder.ActorType,
 				ActorID:   reminder.ActorID,
 			}); derr != nil {
 				log.Errorf("Error deleting reminder %s: %s", reminder.Key(), derr)
+				errCh <- derr
+				return
 			}
+			errCh <- nil
 		}()
-		return ErrReminderCanceled
+
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case errC := <-errCh:
+			if errC != nil {
+				fmt.Println("--- Error deleting reminder", errC)
+				return errC
+			} else {
+				fmt.Println("--- Deleted reminder", errC)
```

Comment:
I don't think the nil-error `errC` should be printed here.
Are these prints going to be updated to a errorf/debugf?

Reply:
Likely here for debug purposes only